### ISSUE TITLE
debug doesn't work properly.

### ISSUE THREAD

**Issue by asukaminato0721** [kenkangxgwe/lsp-wl#35]
I use this code. And into debug mode.

```mma
fib[1]=1;
fib[2]=1;
fib[3]=2;
fib[4]=3;
fib[5]=5;
fib[x_]:=fib[x]=fib[x-1]+fib[x-2];
Print[fib[30]]
```

![image](https://user-images.githubusercontent.com/30024051/105831987-578d2100-6002-11eb-89e2-e0c0809c287c.png)

In git bash, I get these.

```mma
fib[2]=1;

fib[3]=2;

fib[4]=3;

fib[5]=5;

fib[x_]:=fib[x]=fib[x-1]+fib[x-2];

Print[fib[30]

]
[ERROR 2021-01-26T18:14:59] <|type -> event, event -> stopped, body -> <|reason
-> pause, description -> Cell Evaluated Successfully, threadId ->
Key[id][KernelObject[1, local]]|>|>
```

**Comment by kenkangxgwe** [comment]
Sorry, What do you mean by `git bash`? The code you were running isn't gonna return anything other than a newline symbol `\n` , because Print/Echo/Message are not gonna be revealed in the REPL currently. Could you try something like
```mathematica
fib[2]=1;

fib[3]=2;

fib[4]=3;

fib[5]=5;

fib[x_]:=fib[x]=fib[x-1]+fib[x-2];

fib[30]
```

Thanks

**Comment by asukaminato0721** [comment]
Error message in powershell.

```MMA
DataType`TypeCheck::mispat:
   WolframLanguageServer`Adaptor`GetKernelId[KernelObject[1, local]]
     doesn't match the pattern of "id" in
    WolframLanguageServer`Specification`DapThread.

Export::jsonstrictencoding:
   Expression WolframLanguageServer`Adaptor`GetKernelId cannot be exported as
    RawJSON.
[ERROR 2021-01-27T21:05:32] <|type -> event, event -> process, body -> <|name ->
wolfram.exe, systemProcessId ->
WolframLanguageServer`Adaptor`GetKernelId[KernelObject[1, local]],
isLocalProcess -> True, startMethod -> attach|>|>

Export::jsonstrictencoding:
   Expression WolframLanguageServer`Adaptor`GetKernelId cannot be exported as
    RawJSON.

DataType`TypeCheck::mispat:
   WolframLanguageServer`Adaptor`GetKernelId[KernelObject[1, local]]
     doesn't match the pattern of "id" in
    WolframLanguageServer`Specification`DapThread.

Export::jsonstrictencoding:
   Expression WolframLanguageServer`Adaptor`GetKernelId cannot be exported as
    RawJSON.

General::stop: Further output of Export::jsonstrictencoding
     will be suppressed during this calculation.
[ERROR 2021-01-27T21:05:32] <|type -> event, event -> thread, body -> <|reason
-> wolfram.exe, threadId ->
WolframLanguageServer`Adaptor`GetKernelId[KernelObject[1, local]]|>|>
[INFO  2021-01-27T21:05:32] Configuration Done for Wolfram Language Debugger

Key::invak: The argument KernelObject[1, local] is not a valid Association.
[ERROR 2021-01-27T21:05:32] <|type -> event, event -> stopped, body -> <|reason
-> pause, description -> Cell Evaluated Successfully, threadId ->
Key[id][KernelObject[1, local]]|>|>
[ERROR 2021-01-27T21:05:32] <|type -> response, request_seq -> 4, success ->
True, command -> threads, body -> <|threads ->
WolframLanguageServer`Adaptor`GetThreads[KernelObject[1, local]]|>|>
```

Steps:

Launch `init.wls` in powershell, turn on debug mode. Click the green button

![image](https://user-images.githubusercontent.com/30024051/105995229-a8c11180-60e3-11eb-8ff2-ed0fb01e545f.png)

**Comment by kenkangxgwe** [comment]
> Launch init.wls in powershell

Can you show me the full commend and why that is necessary? In VS Code, did the language server start up automatically?

------

And also what version of Wolfram are you using?

**Comment by asukaminato0721** [comment]
## did the language server startup automatically?

It's kind of strange since the old version can startup automatically, but the new needs to start up manually. (I delete the old one and git clone a new one)

## what version of Wolfram are you using

11.3.0 for Microsoft Windows (64-bit) (March 27, 2018)

Following is a full gif. (file name doesn't matter, I tested.)

![GIF](https://user-images.githubusercontent.com/30024051/106132927-8db3d780-619f-11eb-814d-6bec27debb78.gif)

---

Besides, when I open vscode, It says

```mma
FrontEndObject::notavail: A front end is not available; certain operations require a front end.
```

**Comment by asukaminato0721** [comment]
I find this thing:

![GIF](https://user-images.githubusercontent.com/30024051/106148007-81397a00-61b3-11eb-8900-b3a2021bb3b4.gif)

So lsp did startup in some situations. But I can't figure it out.

**Comment by kenkangxgwe** [comment]
Chatted with @wuyudi , the original issue was caused by wrong pattern matching for `KernelObject`. In 11.3, `KernelObject` is `Parallel`Kernels`kernel`. This is fixed in 42f9d2206e099aee1139dc9d0b0af53393d406ef.

The `FrontEndObject::notavail` error was caused by his `init.m` for wolfram.exe, not related to the language server.

**Comment by asukaminato0721** [comment]
```Parallel`Kernels`kernel```

```mma
```Parallel`Kernels`kernel```
```

**Comment by kenkangxgwe** [comment]
Another issue is fixed by 1ca55ad675ee278b0950dd269090204475f75dcc. Please check out if you want to test.

**Comment by asukaminato0721** [comment]
Conclusion:

Open cmd/PowerShell/bash, type `wolfram.exe`, see if there are warning info. If there are, clear the `init.m` files.

Or do sth like this( add context ) :

```mma
PDL = GeneralUtilities`PrintDefinitionsLocal;
Needs["MaTeX`"];
Needs["Rubi`"];
Needs["CCompilerDriver`GenericCCompiler`"];
CCompilerDriver`$CCompiler = {"Compiler" -> CCompilerDriver`GenericCCompiler`GenericCCompiler, 
   "CompilerInstallation" -> "D:\\tcc", "CompilerName" -> "tcc.exe"};
 ]
```

If there are any code needs frontend, add

```mma
If[MatchQ[$FrontEnd,_FrontEndObject],  (* code needs frontend *)
]
```
